src/surface_conditions/surface_conditions.jl

```diff
@@ -122,6 +128,7 @@ function surface_state_to_conditions(
     interior_z,
     params,
     atmos,
+    sfc_prognostic_values,
```

Comment:
@charleskawczynski I am trying to decrease the number of allocations in this PR to meet the threshold in the flame tests (it is 2% over the threshold for the default case, and lower for others). The problem seems to be that passing sfc_prognostic_values into surface_state_to_conditions causes unnecessary allocations (and I don't see a way to update surface temperature without passing sfc_prognostic_values in). Can you please let me know if you have any ideas for how to prevent extra allocations here or if it's even preventable?

Reply:
It's difficult and not very fruitful to diagnose those issues until this branch is rebased, since that may impact allocations. Can you rebase and fix the git conflicts first? cc me after and I can take a look

Reply:
I believe these functions need to be refactored

Reply:
@charleskawczynski I've now rebased, and the number of allocations has not changed

Reply:
@charleskawczynski what exactly do you think should be refactored with these functions apart from the if statements?

Reply:
It's not a matter of this PR, the function is already allocating and allocations seem to easily change when adjusting this function, which makes me think that it's not being inferred well. @dennisYatunin, can you please help @etmaca5 with the allocations?

One question that I have is: why is this function computing `T` when it's being passed in via `surface_state`? I've seen situations in the past where the compiler